front-end/src/components/Deck.js

```diff
@@ -0,0 +1,50 @@
+import React, {useEffect, useState} from 'react';
+import {PlayIcon} from "@heroicons/react/20/solid";
+
+const Deck = ({deck}) => {
+    const learningProgress = Math.round(deck.learnt / deck.total * 100);
+    const [progress, setProgress] = useState(0);
+
+    useEffect(() => {
+            setTimeout(() => {
+                    if (progress < learningProgress) {
+                        setProgress(progress + 1);
+                    }
+                },
+                6)
+        },
+        [progress]);
```

Comment:
Please add learningProgress as a dependency so it re-renders the page when it changes